### PR TITLE
Fix orca and re-add liblouis

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
Orca was broken because of the `gobject-introspection` issue (56943). Adding it to the `propagatedBuildInputs` makes it runnable again. It can't connect to the d-bus but I guess that should work if you have your other braille tools set up properly.

Also `libluois` was removed as a dependency because it had previously been broken - now it works again and we can use it as a dep here :)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
>  It can't connect to the d-bus but I guess that should work if you have your other braille tools set up properly.

Not sure what you mean by the dbus not working?

Otherwise LGTM.

File: pkgs/applications/misc/orca/default.nix [hunk 1]
```diff
@@ -34,8 +34,8 @@ buildPythonApplication rec {
   ];
 
   propagatedBuildInputs = [
-    # TODO: re-add liblouis when it is fixed
-    pygobject3 pyatspi dbus-python pyxdg brltty speechd gst-python setproctitle
+    gobject-introspection  # Temporary fix, see https://github.com/NixOS/nixpkgs/issues/56943
```
Comment (hedning) [inline pkgs/applications/misc/orca/default.nix]: Any reason adding it to `propagatedBuildInputs` and not `buildInputs`?  (in theory setting `strictDeps = false;` should work, but I haven't had any luck with that)

Comment (jtojnar) [reply pkgs/applications/misc/orca/default.nix]: Yeah, we only need to propagate `pythonDependencies`. And I think disabling `strictDeps` is better than adding build time dependencies to `buildInputs` and pretending `strictDeps = true` works.

Comment (hedning) [reply pkgs/applications/misc/orca/default.nix]: I'm getting the same runtime error when having `strictDeps = false;` though, no idea why. Seeing the same issue with `guake`.

Comment (jtojnar) [reply pkgs/applications/misc/orca/default.nix]: Yay, I already managed to forgot it is not overridable from `buildPythonPackage` https://github.com/NixOS/nixpkgs/pull/59187

WE should probably fix that.

**Comment by hedning** [comment]
Looks like `liblouis` is broken in stable, but the strictDeps fix needs backporting.

File: pkgs/applications/misc/orca/default.nix (hunk 1 above)
Comment (JohnAZoidberg) [reply pkgs/applications/misc/orca/default.nix]: Yeah, a ton of python packages have that same problem :/

**Comment by JohnAZoidberg** [comment]
That's why I split it into two commits :)
Please cherry-pick `d65f602f356b3b81cce14941dffc7e0ae65c90c3` onto stable.

File: pkgs/applications/misc/orca/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/applications/misc/orca/default.nix]: A [fix](https://github.com/NixOS/nixpkgs/pull/60178) allowing to use `strictDeps = false;` is now on master. Please rebase and try that.

**Comment by jtojnar** [comment]
Hmm, I am getting the following in console on Orca startup:

```
** (gst-plugin-scanner:25871): CRITICAL **: 22:34:14.805: Couldn't g_module_open libpython. Reason: /nix/store/kimimvhimclyfzlncpg36zjni3wn70nq-python3-3.7.3/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7m.so: cannot open shared object file: No such file or directory

** (gst-plugin-scanner:25871): CRITICAL **: 22:34:14.809: Couldn't g_module_open libpython. Reason: /nix/store/kimimvhimclyfzlncpg36zjni3wn70nq-python3-3.7.3/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7m.so: cannot open shared object file: No such file or directory
```

Probably unrelated.

**Comment by worldofpeace** [comment]
> Hmm, I am getting the following in console on Orca startup:
> 
> ```
> ** (gst-plugin-scanner:25871): CRITICAL **: 22:34:14.805: Couldn't g_module_open libpython. Reason: /nix/store/kimimvhimclyfzlncpg36zjni3wn70nq-python3-3.7.3/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7m.so: cannot open shared object file: No such file or directory
> 
> ** (gst-plugin-scanner:25871): CRITICAL **: 22:34:14.809: Couldn't g_module_open libpython. Reason: /nix/store/kimimvhimclyfzlncpg36zjni3wn70nq-python3-3.7.3/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7m.so: cannot open shared object file: No such file or directory
> ```
> 
> Probably unrelated.

I've been seeing that a lot for python gtk applications.

File: pkgs/applications/misc/orca/default.nix (hunk 1 above)
Comment (JohnAZoidberg) [reply pkgs/applications/misc/orca/default.nix]: Niceeeee!

**Comment by JohnAZoidberg** [comment]
Alright, works with `strictDeps = false`.
Please backport 507873f352ef4c71a94fedb25b26e8c9a83bc111 to stable.

**Comment by hedning** [comment]
nit: could you change the commit message to match what's being done (eg. set strictDeps = false).

**Comment by JohnAZoidberg** [comment]
Sure, had left it because it still kind of does that. But I see that that's confusing.

**Comment by JohnAZoidberg** [comment]
> Hmm, I am getting the following in console on Orca startup:
> 
> ```
> ** (gst-plugin-scanner:25871): CRITICAL **: 22:34:14.805: Couldn't g_module_open libpython. Reason: /nix/store/kimimvhimclyfzlncpg36zjni3wn70nq-python3-3.7.3/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7m.so: cannot open shared object file: No such file or directory
> 
> ** (gst-plugin-scanner:25871): CRITICAL **: 22:34:14.809: Couldn't g_module_open libpython. Reason: /nix/store/kimimvhimclyfzlncpg36zjni3wn70nq-python3-3.7.3/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7m.so: cannot open shared object file: No such file or directory
> ```
> 
> Probably unrelated.

I think that might be because gst-python doesn't have python as a `buildInput`. You could try that. But yes, seems to be unrelated.

**Comment by hedning** [comment]
Cherry-picked in b12b1e2f42841b47b3d1e2e44078dd5b18f6d78f

**Pull request merged**